module-4/backend/week_two.md

```diff
@@ -1,11 +1,55 @@
 ## Week Two - Module 4 Recap
 
-Fork this respository. Answer the questions to the best of your ability. Try to answer them with limited amount of external research. These questions cover the majority of what we've learned this week. 
+Fork this respository. Answer the questions to the best of your ability. Try to answer them with limited amount of external research. These questions cover the majority of what we've learned this week.
 
-Note: When you're done, submit a PR. 
+Note: When you're done, submit a PR.
 
-TBD  
+1. What is Webpack and why is it useful?
+
+    Webpack is a build tool that puts all of your assets, including Javascript, images, fonts, and CSS, in a dependency graph. Webpack lets you use require() in your source code to point to local files, like images, and decide how they're processed in your final Javascript bundle, like replacing the path with a URL pointing to a CDN.
+
+2. When do you want to use event delegation?
+
+  Event delegation allows us to attach a single event listener, to a parent element, that will fire for all descendants matching a selector, whether those descendants exist now or are added in the future.
+
+3. What's one difference between ES5 and ES6?
+
+  In ES6 you can omit 'function' when setting it; use of 'const' and 'let'
+
+4. What's the deal with semi-colons in JavaScript?
+
+  They are not required
+
+5. How are you using the MVC design pattern in your Quantified Self project?
+
+  So far only in the backend part of the project. There's room for refactoring. We have a Foods Controller and a Foods model.
+
+6. How do you execute raw SQL in node?
+
+  By declaring 'database.raw' before the query and by requiring the database in the file
+
+7. What is CORS?
+
+  Cross-origin resource sharing (CORS) is a mechanism that allows restricted resources (e.g. fonts) on a web page to be requested from another domain outside the domain from which the first resource was served.
+
+8. What are some steps to avoid CORS?
+
+  We can set up CORS permissions for specific sites or for all sites.
 
 #### Review  
 
-TBD
+9. Why do people say "HTTP is stateless"?
+
+  Because it doesn't transfer date between different requests.
+
+10. What is a RESTful API?
+
+  A RESTful API is an application program interface (API) that uses HTTP requests to GET, PUT, POST and DELETE data.
```

Comment:
and it also follows some naming conventions for your routes, controllers, actions, etc.